lib/Sema/TypeCheckAttr.cpp

```diff
@@ -6910,6 +6910,26 @@ void AttributeChecker::visitNonisolatedAttr(NonisolatedAttr *attr) {
           return;
         }
 
+        if (attr->isUnsafe() && type->isSendableType() && var->isStatic() &&
+            var->isLet()) {
+          // 'nonisolated(unsafe)' is redundant for static constant 'Sendable'
+          // struct properties.
+          if (isa<StructDecl>(nominal))
+            diagnoseAndRemoveAttr(
+                attr, diag::nonisolated_unsafe_sendable_struct, type);
+
+          // '(unsafe)' is redundant for static constant 'Sendable' actor
+          // properties.
+          // TODO: Should this apply to only public actors?
```

Comment:
Right, the only case where `nonisolated` is necessary on actor-isolated immutable, `Sendable` constants is when they're accessible outside the module, because the variable is already considered to be `nonisolated` when accessed from within the module.